API.md

```diff
@@ -0,0 +1,552 @@
+| options.timeout | `number` | maximum time the query should run |
+| options.maxNumProviders | `number` | maximum number of providers to find |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Array<PeerInfo>>` |  array of [`PeerInfo`](https://github.com/libp2p/js-peer-info) |
+
+#### Example
+
+```js
+// ...
+const providers = await libp2p.contentRouting.findProviders(cid)
+```
+
+### contentRouting.provide
+
+Iterates over all content routers in parallel, in order to notify it is a provider of the given key.
+
+`libp2p.contentRouting.provide(cid)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| cid | [`CID`](https://github.com/multiformats/js-cid) | cid to provide |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<void>` | Promise resolves once notifications are sent |
+
+#### Example
+
+```js
+// ...
+await libp2p.contentRouting.provide(cid)
+```
+
+### _dht.put
+
+Writes a value to a key in the DHT.
+
+`libp2p._dht.put(key, value, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| key | `String` | key to add to the dht |
+| value | `Buffer` | value to add to the dht |
+| [options] | `Object` | put options |
+| [options.minPeers] | `number` | minimum number of peers required to successfully put (default: closestPeers.length) |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<void>` | Promise resolves once value is stored |
+
+#### Example
+
+```js
+// ...
+const key = '/key'
+const value = Buffer.from('oh hello there')
+
+await libp2p._dht.put(key, value)
+```
+
+### _dht.get
+
+Queries the DHT for a value stored for a given key.
+
+`libp2p._dht.get(key, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| key | `String` | key to get from the dht |
+| [options] | `Object` | get options |
+| [options.timeout] | `number` | maximum time the query should run |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Buffer>` | Value obtained from the DHT |
+
+#### Example
+
+```js
+// ...
+
+const key = '/key'
+const value = await libp2p._dht.get(key)
+```
+
+### _dht.getMany
+
+Queries the DHT for the n values stored for the given key (without sorting).
+
+`libp2p._dht.getMany(key, nvals, options)`
+
+#### Parameters
+
+| Name | Type | Description |
+|------|------|-------------|
+| key | `String` | key to get from the dht |
+| nvals | `number` | number of values aimed |
+| [options] | `Object` | get options |
+| [options.timeout] | `number` | maximum time the query should run |
+
+#### Returns
+
+| Type | Description |
+|------|-------------|
+| `Promise<Array<{from: PeerId, val: Buffer}>>` | Array of records obtained from the DHT |
+
+#### Example
+
+```js
+// ...
+
+const key = '/key'
+const { from, val } = await libp2p._dht.get(key)
+```
+
+### pubsub.getPeersSubscribed
```

Comment:
Commonly being subscribed to something is referred to as being a subscriber. Suggestion: `pubsub.getSubscribers(topic)`?

Reply:
Oh, I agree with you on this. We already have this on all the `pubsub` repos but I will see what I can do